assembl/static2/js/app/main.jsx

```diff
@@ -1,25 +1,71 @@
 import React from 'react';
 import { connect } from 'react-redux';
+import { scrollToElement } from './utils/globalFunctions';
+import { getCurrentPhaseIdentifier } from './utils/timeline';
 import { fetchSynthesis } from './actions/synthesisActions';
-import { fetchIdeas } from './actions/ideasActions';
 import { fetchPosts } from './actions/postsActions';
 import { fetchUsers } from './actions/usersActions';
 import Navbar from './components/common/navbar';
 import Footer from './components/common/footer';
 
 class Main extends React.Component {
+  constructor(props) {
+    super(props);
+    const { debateData } = this.props.debate;
+    const paramsIdentifier = this.props.params.phase || getCurrentPhaseIdentifier(debateData.timeline);
+    const queryIdentifier = this.props.location.query.phase || paramsIdentifier;
+    this.state = {
+      identifier: queryIdentifier
+    };
+    this.displayHeader = this.displayHeader.bind(this);
+  }
   componentWillMount() {
     const { debateId, connectedUserId } = this.props.context;
     this.props.fetchPosts(debateId);
-    this.props.fetchIdeas(debateId);
     this.props.fetchSynthesis(debateId);
     this.props.fetchUsers(debateId, connectedUserId);
   }
+  componentDidMount() {
+    window.addEventListener('scroll', this.displayHeader);
```

Comment:
Please use lodash's debounce or throttle, because otherwise as the browser sends these scroll events very often, running the listener code too often can kill performance and produce lag.